functions/_enhancd_entry_git_root.fish

```diff
@@ -0,0 +1,8 @@
+function _enhancd_entry_git_root
+    if git rev-parse --is-inside-work-tree 2> /dev/null
+       and test (git rev-parse --show-toplevel) != (pwd)
```

Comment:
@gazorby I have nothing against `and`, but if we're going to be using `&&` in some places, I'd prefer to use `&&` everywhere.

Reply:
@jorgebucaran I agree with the syntax consistency, but as `&&` and `||` were introduced for compatibility with POSIX shells and we have separate sources for bash/zsh, I'd prefer to use the plain `and` / `or` fish syntax.